### PR TITLE
Update CI to use Python 3.10 - 3.12 and plain Miniforge

### PR DESCRIPTION
**Pull request by pnuu**
Update CI to use Python versions 3.10, 3.11 and 3.12.

In addition switch from Mambaforge (which is deprecated) to plain Miniforge.

### CODE REVIEW

**Review by adybbroe — APPROVED** [review]
LGTM. But why are the CI tests failing? Any idea?

**Comment by pnuu** [comment]
Lets see if switching from Mambaforge to plain Miniconda helps.

**Comment by mraspaud** [comment]
This will use miniforge, not miniconda, right?

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -33,11 +33,12 @@ jobs:
       - name: Setup Conda Environment
         uses: conda-incubator/setup-miniconda@v3
         with:
-          miniforge-variant: Mambaforge
           miniforge-version: latest
-          use-mamba: true
           python-version: ${{ matrix.python-version }}
           activate-environment: test-environment
+          channels: conda-forge
+          conda-remove-defaults: true
```
Comment (sfinkens) [inline .github/workflows/ci.yaml:40]: Oh, that's neat!

**Review by sfinkens — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (pnuu) [reply .github/workflows/ci.yaml:40]: Borrowed from Satpy (which I apparently also did :see_no_evil: :joy:).

**Pull request merged**